Changelog.md

```diff
@@ -8,6 +8,7 @@ Language Features:
 
 Compiler Features:
  * Code Generator: More efficient overflow checks for multiplication.
+ * Code Generator: Use unchecked arithmetic for the increment in specific for loops.
```

Comment:
If this is only an optimization, then we should phrase it as: Remove redundant overflow check for increment...

Reply:
We'd probably need to make it some kind of optimizer setting then, though (i.e. it'd be easiest if it was a setting in `solidity::frontend::OptimiserSettings`)... doesn't need to be configurable in CLI, but maybe in standard-json... for that the crucial question would be if it should get a specific setting of its own or whether we add something generic like "optimized code generation" to be used for all similar cases like this...